internal/ratelimit/config.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package ratelimit
+
+import "fmt"
+
+// UnaryInboundMiddlewareConfig describes how to configure and construct a
+// unary inbound rate limiter.
+type UnaryInboundMiddlewareConfig struct {
+	// RPS is the maximum requests per second, after which the inbound will
+	// throttle inbound requests with a ResourceExhaustedError of "rate limit
+	// exceeded".
+	RPS int `config:"rps" yaml:"rps"`
+	// BurstLimit determines how much slack the rate limiter will tolerate for
+	// a burst of requests from an idle state before throttling.
+	// The default is 10. A burstLimit of 0 implies the default.
+	// Use "noSlack" to configure a rate limiter without slack.
+	BurstLimit int `config:"burstLimit" yaml:"burstLimit"`
+	// NoSlack configures the rate limiter without any slack, even after idling
+	// indefinitely.
+	NoSlack bool `config:"noSlack" yaml:"noSlack"`
```

Comment:
I agree. I could use a hint about the pattern for testing config parsing, or I can just drop testing the config parsing test.